### DISCUSSION TITLE
How are windows created that float by default?

### DISCUSSION THREAD

**Discussion by NicoElbers** [i3/i3#5893] [Q&A] [answered]
This might be the wrong place to ask, if so please say so, but I am curious as to how some windows get created as floating windows by default. This question is form the point of view that I'd like to use it for my own applications. 

As an example, when I run `vkmark` it creates a floating window by default:
![image](https://github.com/i3/i3/assets/119488797/1ffb594b-74d7-41f8-a036-edd8e0a94d0f)

I've tried to dig through some code and I've found another library [olive.c](https://github.com/tsoding/olive.c/tree/master) which also seems to have [floating windows through SDL](https://github.com/tsoding/olive.c/blob/c0eb82c54d69f6f8eefc4f1f33f3459a664eb529/demos/vc.c#L76) (found in the demos):
![image](https://github.com/i3/i3/assets/119488797/b982ec21-3e80-4251-9a17-93fe5f4706b1)

But digging through the SDL source code lead to [dead ends](https://github.com/libsdl-org/SDL/blob/4a759b340ff3addaaa1ee8382f4ae81c45d17c97/src/video/qnx/SDL_qnxvideo.c#L82) (I was unable to find the code responsible for actually creating the window).

Mind you, I did not change my i3 config for any of this. The only references to "floating" in my config are for moving and toggling floating windows. 

So my question is, if I want to create my own application that will I guess hint to i3 that it wants to be floating, how do I do that?

**Comment by orestisfl** [comment, marked as answer]
You can take a look in i3's code here: https://github.com/i3/i3/blob/230147c81547848c83ac7764a6c1691f26f10e05/src/manage.c#L464-L476

By default, windows with "dialog" hints or transient windows become floating

**Reply by NicoElbers** [reply]
Thank you very much, I'll have a look later today!

**Reply by NicoElbers** [reply]
That did the trick! Thank you very much!
